docs/specification.md

```diff
@@ -0,0 +1,169 @@
+
+1. Structs
+
+    - Prefix with Editor: For structs related to the core editor functionality.
+    - Prefix with Command: For structs related to commands.
+    - Prefix with File: For structs related to file operations.
+    - Prefix with Text: For structs related to text manipulation.
+    - Prefix with Render: For structs related to rendering.
+    - Prefix with Syntax: For structs related to syntax highlighting.
+
+2. Files
+
+    - mod.rs: For module files that re-export other modules.
+    - file_commands.rs: For file-related commands.
+    - edit_commands.rs: For text editing commands.
+    - text_buffer.rs: For text buffer management.
+    - renderer.rs: For rendering the text buffer.
+    - file_manager.rs: For file operations.
+    - syntax_highlighter.rs: For syntax highlighting.
+
+3. Functions
+
+    - Prefix with handle_: For functions that handle specific actions.
+    - Prefix with render_: For functions that render content.
+    - Prefix with process_: For functions that process input or commands.
+    - Prefix with load_: For functions that load data.
+    - Prefix with save_: For functions that save data.
+    - Prefix with apply_: For functions that apply changes.
+
+4. Traits
+
+    - Prefix with EditorTrait: For traits related to the core editor functionality.
+    - Prefix with CommandTrait: For traits related to commands.
+    - Prefix with FileTrait: For traits related to file operations.
+    - Prefix with TextTrait: For traits related to text manipulation.
+    - Prefix with RenderTrait: For traits related to rendering.
+    - Prefix with SyntaxTrait: For traits related to syntax highlighting.
+
+5. Constants
+
+    - Prefix with MAX_: For maximum values.
+    - Prefix with MIN_: For minimum values.
+    - Prefix with DEFAULT_: For default values.
+    - Prefix with ERROR_: For error codes or messages.
+
+6. Macros
+
+    - Prefix with macro_: For general-purpose macros.
+    - Prefix with log_: For logging-related macros.
+    - Prefix with assert_: For assertion-related macros.
+    - Prefix with debug_: For debugging-related macros.
+
+## Versioning ! 
+> using SemVer !   
+### Semantic Versioning (SemVer)
+
+- SemVer defines three components of a version number:
+
+    - MAJOR: Incremented for incompatible API changes.
+    - MINOR: Incremented for adding functionality in a backwards-compatible manner.
+    - PATCH: Incremented for backwards-compatible bug fixes.
+
+- Additionally, pre-release versions and build metadata can be appended to the version number:
+
+    - Pre-release: Denoted by a hyphen and a series of dot-separated identifiers (e.g., 1.0.0-alpha, 1.0.0-beta.2).
+    - Build metadata: Denoted by a plus sign and a series of dot-separated identifiers (e.g., 1.0.0+build.123).
+
+- When to Increment Versions
+
+    - MAJOR Version (e.g., 1.0.0 to 2.0.0):
+        - Increment when you make incompatible API changes.
+        - This indicates that users may need to update their code to use the new version.
```

Comment:
```suggestion
        - This indicates that users may need to update their text editor to use the new version.
```

Reply:
nn here is code , it's related to the development rather than the release !